### PR TITLE
Add isGeneric support

### LINKED ISSUE #6: Add reflection to types to know when they are generic
We need to add ability to recognize generic types, as often we want to avoid those when doing code generation. 

E.g. we'll want to skip `allValues` generation for Enums that are generic.

### PR DESCRIPTION
**Pull request by krzysztofzablocki**
@ilyapuchka Since you have been so helpful, I've invited you to the project as a collaborator, hope that's ok :) 
Check this PR out

Closes #6

### CODE REVIEW

File: Sourcery/Models/Type.swift [hunk 1]
```diff
@@ -14,10 +14,15 @@ class Type: NSObject {
...
     /// Name in global scope 
     var name: String {
-        guard let parentName = parentName else { return localName }
+        guard let parentName = parent?.name else { return localName }
         return "\(parentName).\(localName)"
     }
 
+    /// Is this type generic? regardless whether it's this type directly or one of its parent types
+    var isGeneric: Bool {
+        return (parent?.isGeneric ?? false) || hasGenericComponent
```
Comment (krzysztofzablocki) [inline Sourcery/Models/Type.swift]: Before Swift 3 they couldn't, now they can

Comment (ilyapuchka) [inline Sourcery/Models/Type.swift]: Really? 😯 I got a compilation error with this code:
```
struct MyStruct<Element> {
    struct Inner { } //Type 'Inner' can not be nested in generic type 'MyStruct'
}
```

Comment (krzysztofzablocki) [inline Sourcery/Models/Type.swift]: Do you think we should remove this check to improve performance, or doesn't matter?

Comment (krzysztofzablocki) [inline Sourcery/Models/Type.swift]: ok, I'll get rid of it for now 👍

**Pull request merged**